src/app/handlers/acknowledge.handler.ts

```diff
@@ -1,21 +1,20 @@
 import { GuildChannel, MessageReaction, User, Role } from 'discord.js';
 import { IContainer, IHandler, Maybe } from '../../common/types';
 import Constants from '../../common/constants';
+import { MemberUtils } from '../util/member.util';
 
 export class AcknowledgeHandler implements IHandler {
   private _CoC_Channel: Maybe<GuildChannel> = undefined;
-  private _UNACKNOWLEDGED_ROLE: string = 'Un Acknowledged';
-  private _roleCache: Record<string, Maybe<Role>> = {
-    [this._UNACKNOWLEDGED_ROLE]: undefined,
-  };
+  private _TARGET_ROLE_NAME: string = 'Un Acknowledged';
+  private _UNACKNOWLEDGED_ROLE: Maybe<Role> = undefined;
 
   constructor(public container: IContainer) {}
 
   public async execute(reaction: MessageReaction, user: User): Promise<void> {
-    if (!this._roleCache[this._UNACKNOWLEDGED_ROLE]) {
-      this._roleCache[this._UNACKNOWLEDGED_ROLE] = this.container.guildService
+    if (!this._UNACKNOWLEDGED_ROLE) {
+      this._UNACKNOWLEDGED_ROLE = this.container.guildService
```

Comment:
if we do this pattern often, maybe make a `GuildService::getRole(r: string)`

Reply:
then the guildservice can maintain a single "role cache" and either serve from said cache or do a lookup